vars/provisionNodes.groovy

```diff
@@ -121,10 +121,17 @@ def call(Map config = [:]) {
                                              python-debuginfo python2-aexpect libcmocka          \
                                              python-pathlib python2-numpy git                    \
                                              golang-bin'''
-    sh script: 'set -x; rm -f ci_key*; ssh-keygen -N "" -f ci_key;' +
-               ' pdcp -R ssh -l root -w ' + nodeString +
-               ' ci_key* /tmp/;' +
-               ' pdsh -R ssh -l root -w ' + nodeString +
-               ' "' + provision_script + '" 2>&1 | dshbak -c'
+    def rc = 0
+    rc = sh(script: 'set -x; rm -f ci_key*; ssh-keygen -N "" -f ci_key;' +
```

Comment:
I find it much clearer to read the cart/Jenkinsfile script where """ is used rather than '.  Even if this way is preferred, you could use ' pdcp -R ssh -l root -w $nodeString' + to do variable substitution.

Reply:
Yeah.  It can just sometimes lead down escaping hell, particularly when you have shell script that needs escaping inside of groovy script that needs escaping, etc.

In this case I don't think it's so bad, so I have changed to multi-line strings using `"""` and '`'''` as needed.